app/javascript/components/pages/EventShow.js

```diff
@@ -1,6 +1,40 @@
 import React, { Component } from 'react';
 
 class EventShow extends Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      loading: true,
+      event: false,
+    }
+  }
+//
+componentDidMount(){
+  this.eventShow()
+}
+
+  eventShow = (props) => {
+    fetch(`http://localhost:3000/parties/${this.props.id}`)
+      .then((response) => {
+        if(response.status === 200){
+          return response.json();
+        }
+      })
+      .then((singleEvent) => {
+        this.setState({event: singleEvent});
+        this.setState({loading: false})
```

Comment:
generally it's best practice to reduce the number of times setState gets called because it has asynchronous behavior. Here you can call it once but pass it an object with multiple properties
```suggestion
        this.setState({event: singleEvent, loading: false});
```